### PR TITLE
Update to sylius 1 10

### PR DESCRIPTION
**Pull request by pszost**

### CODE REVIEW

File: behat.yml.dist [hunk 1]
```diff
@@ -4,19 +4,25 @@ imports:
...
+
+        FriendsOfBehat\MinkDebugExtension:
             directory: etc/build
             clean_start: false
             screenshot: true
 
         Behat\MinkExtension:
             files_path: "%paths.base%/vendor/sylius/sylius/src/Sylius/Behat/Resources/fixtures/"
-            base_url: "http://localhost:8080/"
+            base_url: "https://127.0.0.1:8080/"
```
Comment (er1z) [inline behat.yml.dist]: Are you sure? Having required a SSL connection provides some additional effort to run tests.

File: composer.json [hunk 2]
```diff
@@ -2,65 +2,64 @@
...
-        "symfony/browser-kit": "4.1.8",
-        "symfony/dependency-injection": "4.1.8",
-        "symfony/dom-crawler": "4.1.8",
-        "symfony/routing": "4.1.8"
+    "config": {
+        "sort-packages": true
+    },
+    "extra": {
+        "branch-alias": {
+            "dev-master": "1.10-dev"
```
Comment (er1z) [inline composer.json]: Do we want it?

File: composer.json [hunk 3]
```diff
@@ -2,65 +2,64 @@
...
+        }
     },
-    "prefer-stable": true,
```
Comment (er1z) [inline composer.json:56]: Why?

File: src/EventListener/ShippingExportEventListener.php [hunk 4]
```diff
@@ -14,35 +14,64 @@
...
-        $shippingExport = $exportShipmentEvent->getShippingExport();
+        /** @var ShippingExportInterface $shippingExport */
+        $shippingExport = $event->getSubject();
+        Assert::isInstanceOf($shippingExport, ShippingExportInterface::class);
+
         $shippingGateway = $shippingExport->getShippingGateway();
+        Assert::notNull($shippingGateway);
 
-        if ($shippingGateway->getCode() !== self::DHL_GATEWAY_CODE) {
+        if ('dhl24_pl' !== $shippingGateway->getCode()) {
```
Comment (er1z) [inline src/EventListener/ShippingExportEventListener.php]: To const.

File: src/EventListener/ShippingExportEventListener.php [hunk 5]
```diff
@@ -51,23 +80,68 @@ public function exportShipment(ExportShipmentEvent $exportShipmentEvent): void
...
 
         $labelContent = base64_decode($response->createShipmentResult->label->labelContent);
-        $extension = self::BASE_LABEL_EXTENSION;
 
-        if ('ZBLP' === $response->createShipmentResult->label->labelType) {
-            $extension = 'zpl';
-        }
+        $this->flashBag->add('success', 'bitbag.ui.shipment_data_has_been_exported'); // Add success notification
+        $this->saveShippingLabel($shippingExport, $labelContent, 'pdf'); // Save label
+        //$this->markShipmentAsExported($shippingExport); // Mark shipment as "Exported"
```
Comment (er1z) [inline src/EventListener/ShippingExportEventListener.php]: Is this comment required?

File: src/EventListener/ShippingExportEventListener.php [hunk 6]
```diff
@@ -14,35 +14,64 @@
...
 
-    public function __construct(WebClientInterface $webClient, SoapClientInterface $soapClient)
-    {
+    public function __construct(
+        FlashBagInterface $flashBag,
+        Filesystem $filesystem,
+        ObjectManager $shippingExportManager,
+        string $shippingLabelsPath,
+        WebClientInterface $webClient,
+        SoapClientInterface $soapClient
```
Comment (er1z) [inline src/EventListener/ShippingExportEventListener.php]: Is it possible to break this listener into separate classes? 6 dependencies is a pretty high number. Once you try to write a unit test for this class, it'll be intuitive.

File: tests/Application/config/routes.yaml [hunk 7]
```diff
@@ -1,3 +1,3 @@
-bitbag_sylius_shipping_export_plugin:
```
Comment (er1z) [inline tests/Application/config/routes.yaml:1]: Should empty configuration files be present in the repository?

Comment (pszost) [reply tests/Application/config/routes.yaml:1]: It is not empty. It contains

```
bitbag_shipping_export_plugin:
    resource: "@BitBagSyliusShippingExportPlugin/Resources/config/routing.yml"
    prefix: /admin

```

File: src/Api/ShippingLabelFetcher.php [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    public function createShipment($shippingGateway, $shipment): void
+    {
+        $this->webClient->setShippingGateway($shippingGateway);
+        $this->webClient->setShipment($shipment);
+
+        try {
+            $requestData = $this->webClient->getRequestData();
+
+            $this->response = $this->soapClient->createShipment($requestData, $shippingGateway->getConfigValue('wsdl'));
+        } catch (\Exception $exception) {
```
Comment (er1z) [inline src/Api/ShippingLabelFetcher.php]: Shouldn't be so wide? Why not handle a particular exception or provide a list of handled exception here?

File: src/EventListener/ShippingExportEventListener.php [hunk 9]
```diff
@@ -12,62 +12,108 @@
...
+            . '.' . $labelExtension;
 
-        if ('ZBLP' === $response->createShipmentResult->label->labelType) {
-            $extension = 'zpl';
-        }
+        $this->filesystem->dumpFile($labelPath, $labelContent);
+        $shippingExport->setLabelPath($labelPath);
+
+        $this->shippingExportManager->persist($shippingExport);
+        $this->shippingExportManager->flush();
```
Comment (er1z) [inline src/EventListener/ShippingExportEventListener.php]: Sylius base repository handles both of above.

File: tests/Application/config/routes.yaml (hunk 7 above)
Comment (er1z) [reply tests/Application/config/routes.yaml:1]: I know this is not empty, but there other configuration files in this directory that are simply empty.

File: src/Api/ShippingLabelFetcher.php [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        $this->flashBag = $flashBag;
+        $this->webClient = $webClient;
+        $this->soapClient = $soapClient;
+    }
+
+    public function createShipment($shippingGateway, $shipment): void
+    {
+        $this->webClient->setShippingGateway($shippingGateway);
+        $this->webClient->setShipment($shipment);
```
Comment (er1z) [inline src/Api/ShippingLabelFetcher.php]: Forgot previously; shouldn't this also be within the catch block?

**Pull request merged**